infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -0,0 +1,198 @@
+package org.corfudb.infrastructure.log;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.ResourceQuota;
+import org.corfudb.infrastructure.ServerContext;
+import org.corfudb.infrastructure.log.FileSystemAgent.PartitionAgent.PartitionAttribute;
+import org.corfudb.runtime.exceptions.LogUnitException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+@Slf4j
+public final class FileSystemAgent {
+    private static Optional<FileSystemAgent> instance = Optional.empty();
+
+    private final FileSystemConfig config;
+    // Resource quota to track the log size
+    @Getter
+    private final ResourceQuota logSizeQuota;
+
+    @Getter
+    private final PartitionAttribute partitionAttribute;
+
+    private FileSystemAgent(FileSystemConfig config) {
+        this.config = config;
+
+        long initialLogSize = estimateSize();
+        long logSizeLimit = getLogSizeLimit();
+
+        logSizeQuota = new ResourceQuota("LogSizeQuota", logSizeLimit);
+        partitionAttribute = new PartitionAgent(config).getPartitionAttribute();
+
+        logSizeQuota.consume(initialLogSize);
+        log.info("StreamLogFiles: {} size is {} bytes, limit {}", config.logDir, initialLogSize, logSizeLimit);
```

Comment:
Can this log line be updated to reflect `FileSystemAgent` instead of `StreamLogFiles`?

Reply:
done